tortoise/ninja_tortoise.go

```diff
@@ -537,3 +538,17 @@ func (ni *ninjaTortoise) handleIncomingLayer(newlyr *types.Layer) { //i most rec
 	ni.Info("finished layer %d pbase is %d", newlyr.Index(), ni.pBase.Layer())
 	return
 }
+
+func (ni *ninjaTortoise) GetGoodPattern(layer types.LayerID) (uint32, error) {
```

Comment:
Is it thread safe? @almogdepaz

Reply:
nope, we need to add a lock

Reply:
@gavraz can you add a mutex there ?  
talk with me if you need help with this, or if you prefer i do it

Reply:
Added lock.
@almogdepaz please review

Reply:
looks good to me